### PR TITLE
feat: handles Zone Control Plane `zone.enabled` field

### PR DESCRIPTION
**Pull request by kleinfreund**
Treats a Zone Control Plane's `zone.enabled` field as a signal to show its status as "disabled" when its value is `false`.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: src/app/main-overview/components/ZoneControlPlanesDetails.vue [hunk 1]
```diff
@@ -55,7 +55,13 @@ const props = defineProps({
...
   const { name } = zoneOverview
 
-  const status = getItemStatusFromInsight(zoneOverview.zoneInsight)
+  let status: StatusKeyword | 'disabled'
+
+  if (zoneOverview.zone.enabled === false) {
+    status = 'disabled'
+  } else {
+    status = getItemStatusFromInsight(zoneOverview.zoneInsight)
+  }
```
Comment (slonka) [inline src/app/main-overview/components/ZoneControlPlanesDetails.vue:64]: nitpick: if I read it correctly and by default it's disabled then why do we need to set it to disabled one more time? can we just do:

```suggestion
  if (zoneOverview.zone.enabled === true) {
    status = getItemStatusFromInsight(zoneOverview.zoneInsight)
  }
```

File: src/app/zones/views/ZoneListView.vue [hunk 2]
```diff
@@ -252,7 +252,13 @@ function transformToTableData(zoneOverviews: ZoneOverview[]): ZoneOverviewTableR
...
     })
 
-    const status = getItemStatusFromInsight(zoneOverview.zoneInsight)
+    let status: StatusKeyword | 'disabled'
+
+    if (zoneOverview.zone.enabled === false) {
+      status = 'disabled'
+    } else {
+      status = getItemStatusFromInsight(zoneOverview.zoneInsight)
+    }
```
Comment (slonka) [inline src/app/zones/views/ZoneListView.vue:261]: same as above

**Review by slonka — APPROVED** [review]
just 2 nits

File: src/app/main-overview/components/ZoneControlPlanesDetails.vue (hunk 1 above)
Comment (kleinfreund) [reply src/app/main-overview/components/ZoneControlPlanesDetails.vue]: There is no default value in this context currently. Perhaps you read the type annotation above wrong?

Current:

```ts
  let status: StatusKeyword | 'disabled'

  if (zoneOverview.zone.enabled === false) {
    status = 'disabled'
  } else {
    status = getItemStatusFromInsight(zoneOverview.zoneInsight)
  }
```

We can however do what you suggest:

```ts
  let status: StatusKeyword | 'disabled' = 'disabled'

  if (zoneOverview.zone.enabled === true) {
    status = getItemStatusFromInsight(zoneOverview.zoneInsight)
  }
```

I made those changes in the two relevant places now.

File: src/app/zones/views/ZoneListView.vue (hunk 2 above)
Comment (kleinfreund) [reply src/app/zones/views/ZoneListView.vue]: Addressed in https://github.com/kumahq/kuma-gui/pull/1297#discussion_r1289659869.

**Pull request merged**